src/main/java/com/ibm/eventstreams/connect/mqsource/builders/BaseRecordBuilder.java

```diff
@@ -74,18 +77,24 @@ else if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_J
             }
         }
 
-        log.trace("[{}]  Exit {}.configure", Thread.currentThread().getId(), this.getClass().getName());
-    }
+		String str = props.get(MQSourceConnector.CONFIG_NAME_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER);
+
+		copyJmsPropertiesFlag = Boolean.parseBoolean(str);
```

Comment:
Defensive coding style please. Check str is null before using it. Since copyJmsPropertiesFlag is defaulting to false, you don't need to assign it if in the case that str is null.

Reply:
@AndrewJSchofield  Isn't  below code block in MQSourceConnector going to provide a default value(false) if null was assigned to config parameter.
com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java:288

```
config.define(CONFIG_NAME_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER, Type.BOOLEAN, Boolean.FALSE, Importance.LOW,
                      CONFIG_DOCUMENTATION_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER, CONFIG_GROUP_MQ, 20, Width.MEDIUM,
                      CONFIG_DISPLAY_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER);
```

I am anyway going to make suggested changes.